packages/turf-dissolve/index.js

```diff
@@ -0,0 +1,161 @@
+var turfUnion = require('@turf/union');
+var turfOverlaps = require('turf-overlaps');
+var turfbbox = require('@turf/bbox');
+var Rbush = require('rbush');
+var gju = require('geojson-utils');
+var getClosest = require('get-closest');
+
+/**
+ * Dissolves a FeatureCollection of polygons based on a property. Note that multipart features within the collection are not supported
+ *
+ * @name dissolve
+ * @param {FeatureCollection<Polygon>} featureCollection input feature collection to be dissolved
+ * @param {string} propertyName property name on which to dissolve features
+ * @return {FeatureCollection} a FeatureCollection containing the dissolved polygons
+ * @example
+ * var features = {
+ * "type": "FeatureCollection",
+ * "features": [
+ *   {
+ *     "type": "Feature",
+ *     "properties": {
+ *       "combine": "yes"
+ *     },
+ *     "geometry": {
+ *       "type": "Polygon",
+ *       "coordinates": [[[0, 0], [0, 1], [1, 1], [1, 0], [0, 0]]]
+ *     }
+ *   },
+ *   {
+ *     "type": "Feature",
+ *     "properties": {
+ *       "combine": "yes"
+ *     },
+ *     "geometry": {
+ *       "type": "Polygon",
+ *       "coordinates": [[[0, -1], [0, 0], [1, 0], [1, -1], [0,-1]]]
+ *     }
+ *   },
+ *   {
+ *     "type": "Feature",
+ *     "properties": {
+ *       "combine": "no"
+ *     },
+ *     "geometry": {
+ *       "type": "Polygon",
+ *       "coordinates": [[[1,-1],[1, 0], [2, 0], [2, -1], [1, -1]]]
+ *     }
+ *   }
+ *  ]
+ * }
+ *
+ * var dissolved = turf.dissolve(features, 'combine');
+ *
+ * //=dissolved
+ */
+module.exports = function (featureCollection, propertyName) {
+
+    var originalIndexOfItemsRemoved = [];
+    var treeItems = [];
+    var rtree = new Rbush();
+    for (var polyIndex = 0; polyIndex < featureCollection.features.length; polyIndex++) {
+        var inputFeatureBbox = turfbbox(featureCollection.features[polyIndex]);
+        var treeObj = {
+            minX: inputFeatureBbox[0],
+            minY: inputFeatureBbox[1],
+            maxX: inputFeatureBbox[2],
+            maxY: inputFeatureBbox[3],
+            origIndexPosition: polyIndex
+        };
+        treeItems.push(treeObj);
+    }
+    rtree.load(treeItems);
+
+    for (var i = 0; i < featureCollection.features.length; i++) {
```

Comment:
Another alternative is to use the array's `forEach` method. I find it to be expressive like the former but offers the option to pass in the index like the latter. Since you are not breaking out of the loop and only using `continue` shouldn't be an issue using `forEach`.